File: projecteuler/Euler006.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int t = scanner.nextInt();
    long n, sumOfSquares, squareOfTheSum;

    for(int i = 0; i < t; i++) {
      n = scanner.nextInt();
      sumOfSquares = (n * (n + 1) * (2 * n + 1)) / 6;
      squareOfTheSum = ((n * (n + 1)) * (n * (n + 1))) / 4;
      System.out.println(squareOfTheSum - sumOfSquares);
    }
  }
}

File: projecteuler/euler006.rb
gets.to_i.times do
  n = gets.to_i
  sum_of_squares = (n * (n + 1) * (2 * n + 1)) / 6
  square_of_the_sum = ((n * (n + 1)) * (n * (n + 1))) / 4
  puts square_of_the_sum - sum_of_squares
end
